packages/taco/test/conditions/context.test.ts

```diff
@@ -116,13 +120,31 @@ describe('context', () => {
       );
     });
 
-    it('detects if a signer is required', () => {
+    it('detects when signer is required by parameters', () => {
       const conditionObj = {
         ...testContractConditionObj,
+        parameters: [USER_ADDRESS_PARAM],
         returnValueTest: {
-          ...testReturnValueTest,
+          comparator: '==',
+          value: 100,
+        } as ReturnValueTestProps,
+      };
+      const condition = new ContractCondition(conditionObj);
+      const conditionExpr = new ConditionExpression(condition);
+      expect(conditionExpr.contextRequiresSigner()).toBe(true);
+      expect(conditionExpr.buildContext(provider, {}, signer)).toBeDefined();
+      expect(() => conditionExpr.buildContext(provider, {})).toThrow(
+        `Signer required to satisfy ${USER_ADDRESS_PARAM} context variable in condition`,
+      );
+    });
+
+    it('detects when signer is required by return value test', () => {
+      const conditionObj = {
+        ...testContractConditionObj,
+        returnValueTest: {
+          comparator: '==',
```

Comment:
Can we do something like the following to have the user address comparator value make sense, since `testContractCOnditionObj` is a ERC20.balanceOf condition that returns an int? 

```suggestion
      const conditionObj = {
        ...testContractConditionObj,
        standardContractType: 'ERC721',
        method: 'ownerOf',
        parameters: [3591],
        returnValueTest: {
          comparator: '==',
```